dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -145,16 +145,21 @@
 
 {%- endmacro %}
 
-{% macro safe_create_table_as(temporary, relation, sql) -%}
+{% macro safe_create_table_as(temporary, relation, sql, batch_inserts) -%}
     {%- if temporary -%}
       {%- do run_query(create_table_as(temporary, relation, sql, True)) -%}
```

Comment:
I think we need to force batching here as well. If source data is huge, this query may fail

Reply:
good catch, we should allow that also for temporary tables.

Reply:
Yeah, it's a good idea to add batching inserts for non partitions tables.
Could you please create issue for this?

Reply:
I agree with @Danya-Fpnk here,let's narrow down to cases where we have partitioned datasets.
For huge datasets users might want to consider even another approach like insert_by_period.

Reply:
I gave a 2nd thought to this and I changed my mind, I agree with what @svdimchenko raised. The new feature should allow the use the "force" batch behavior everywhere, even for temporary tables.

Copying from my conversation with @svdimchenko this can be as easy as adding this:
```
if temporary
  if force_create_batch:
    create_table_as_with_partitions
  else
    create_unpartitioned
```

Reply:
Yes, great suggestion.
Today I will add this.

Reply:
Yes, great suggestion!

Reply:
it looks like reslved for now, great job done @Danya-Fpnk 💪